ddtrace/contrib/responder/patch.py

```diff
@@ -0,0 +1,109 @@
+
+# stdlib
+
+# project
+import ddtrace
+from ddtrace.ext import http
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+from ddtrace.utils.wrappers import unwrap as _u
+from ddtrace.propagation.http import extract_context_from_http_headers
+
+# 3p
+import responder
+patches = []  # will be overriden when patched funcs are defined.
+
+
+def patch():
+    if _is_patched(responder):
+        return
+    _set_patched(responder, True)
+
+    for obj, mname, fname, func in patches:
+        _w(mname, fname, func)
+
+
+def unpatch():
+    if not _is_patched(responder):
+        return
+    _set_patched(responder, False)
+
+    for obj, _, prop, _ in patches:
+        if '.' in prop:
+            attr, _, prop = prop.partition('.')
+            obj = getattr(obj, attr, object())
+        _u(obj, prop)
+
+
+async def _api_call(wrapped, instance, args, kwargs):
+    tracer = _get_tracer(instance)
+    assert tracer
+
+    scope, receive, send = args  # FIXME: what if this is partially using kwargs?
+
+    # check for propagated traced ids in the http heaers.
+    headers = {k.decode(): v for k, v in scope.get('headers', ())}
+    ctx = extract_context_from_http_headers(headers)
+    if ctx.trace_id:
+        tracer.context_provider.activate(ctx)
+
+    with tracer.trace("responder.request", service="responder") as span:
+        traced_send = _trace_asgi_send_func(send, span)
+        span.set_tag(http.METHOD, scope.get('method'))
+
+        await wrapped(scope, receive, traced_send)
+
+        span.resource = scope.get('__dd_route', '404')
+
+
+def _api_template(wrapped, instance, args, kwargs):
+    # FIXME: needs tests
+    tracer = _get_tracer(instance)
+    with tracer.trace("responder.render_template") as span:
+        span.set_tag("template", args[0])  # FIXME args or kwargs?
```

Comment:
`args[0]` is the `filename` positional argument. Being the underlying `jinja2` calls will get traced, adding this tag might be redundant metadata:

https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/jinja2/patch.py#L52-L57